akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala

```diff
@@ -136,52 +66,213 @@ abstract class PathDirectives extends MiscDirectives {
    * For further information, refer to:
    * [[http://googlewebmastercentral.blogspot.de/2010/04/to-slash-or-not-to-slash.html]]
    */
-  def pathEndOrSingleSlash: Directive = RawPathPrefixForMatchers(List(PathMatchers.SLASH.optional, PathMatchers.END))
+  def pathEndOrSingleSlash(inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathEndOrSingleSlash { inner.get.toScala }
+  }
 
   /**
    * Only passes on the request to its inner route if the request path
    * consists of exactly one remaining slash.
    */
-  def pathSingleSlash: Directive = RawPathPrefixForMatchers(List(PathMatchers.SLASH, PathMatchers.END))
+  def pathSingleSlash(inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathSingleSlash { inner.get.toScala }
+  }
+
+  /**
+   * Applies the given [[PathMatcher]] to the remaining unmatched path after consuming a leading slash.
+   * The matcher has to match the remaining path completely.
+   * If matched the value extracted by the [[PathMatcher]] is extracted on the directive level.
+   */
+  def path(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.path(segment) { inner.get.toScala }
+  }
+  def path(inner: java.util.function.Function[String, Route]): Route = ScalaRoute {
+    D.path(PathMatchers.Segment) { element ⇒ inner.apply(element).toScala }
+  }
+  def path(p: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.path(p.toScala) { inner.get.toScala }
+  }
+  def path[T](p: PathMatcher1[T], inner: JFunction[T, Route]): Route = ScalaRoute {
+    D.path(p.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def path[T1, T2](p: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.path(p.toScala) { (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Applies the given [[PathMatcher]] to a prefix of the remaining unmatched path after consuming a leading slash.
+   * The matcher has to match a prefix of the remaining path.
+   * If matched the value extracted by the PathMatcher is extracted on the directive level.
+   */
+  def pathPrefix(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathPrefix(segment) { inner.get.toScala }
+  }
+  def pathPrefix(inner: java.util.function.Function[String, Route]): Route = ScalaRoute {
+    D.pathPrefix(PathMatchers.Segment) { element ⇒ inner.apply(element).toScala }
+  }
+  def pathPrefix(p: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathPrefix(p.toScala) { inner.get.toScala }
+  }
+  def pathPrefix[T](p: PathMatcher1[T], inner: JFunction[T, Route]): Route = ScalaRoute {
+    D.pathPrefix(p.toScala) { t1 ⇒ inner.apply(t1).toScala }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/vasseurjb/akka/file/2046270054/issues/source?bid=3132306&fileBranchId=3132379#l117)